rasa/server.py

```diff
@@ -377,6 +377,128 @@ def validate_request_body(request: Request, error_message: Text) -> None:
+                    }
+                },
+                {
+                    "properties": {
+                        "event": {"const": "entities"},
+                        "entities": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "start": {"type": "integer"},
+                                    "end": {"type": "integer"},
+                                    "name": {"type": "string"},
+                                    "confidence": {"type": "number"},
+                                    "extractor": {"type": "string"},
+                                    "value": {"type": "object"},
+                                },
+                            },
+                        },
+                    }
+                },
+                {"properties": {"event": {"const": "user_featurization"}}},
+                {"properties": {"event": {"const": "cancel_reminder"}}},
+                {"properties": {"event": {"const": "reminder"}}},
+                {"properties": {"event": {"const": "action_execution_rejected"}}},
+                {"properties": {"event": {"const": "form_validation"}}},
+                {"properties": {"event": {"const": "loop_interrupted"}}},
+                {"properties": {"event": {"const": "form"}}},
+                {"properties": {"event": {"const": "active_loop"}}},
+                {"properties": {"event": {"const": "reset_slots"}}},
+                {"properties": {"event": {"const": "resume"}}},
+                {"properties": {"event": {"const": "pause"}}},
+                {"properties": {"event": {"const": "followup"}}},
+                {"properties": {"event": {"const": "export"}}},
+                {"properties": {"event": {"const": "restart"}}},
+                {"properties": {"event": {"const": "undo"}}},
+                {"properties": {"event": {"const": "rewind"}}},
+                {"properties": {"event": {"const": "bot"}}},
+                {"properties": {"event": {"const": "session_started"}}},
+                {"properties": {"event": {"const": "agent"}}},
+            ],
+        },
+    }
+
+
+def validate_events_in_request_body(request: Request) -> None:
+    """Validates events format in request body."""
+    from jsonschema import validate
+    from jsonschema import ValidationError
+
+    if not isinstance(request.json, list):
+        events = [request.json]
+    else:
+        events = request.json
+
+    try:
+        validate(events, events_request_format_spec())
+    except ValidationError as e:
+        e.message += (
+            " Failed to validate the events format in the action server request body."
+            "For more information about the format visit: "
+            "https://rasa.com/docs/action-server/pages/action-server-api"
+        )
+        raise e
```

Comment:
we should raise an `ErrorResponse` with status code 400. This will currently result in a 500 error code